src/api.ts

```diff
@@ -582,17 +582,19 @@ async function injectAndRecordTx(
 }> {
   const { raw } = tx
   let nodeIpPort: string, baseUrl: string
-
-  // Initialize them with some values({ nodeIpPort, baseUrl } = getBaseUrl())
+  ;({ nodeIpPort, baseUrl } = getBaseUrl())
 
   if (config.enableBlacklistingIP) {
-    // Continuously reassign nodeIpPort and baseUrl until a non-blacklisted one is found
     let entry
 
     do {
-      ;({ nodeIpPort, baseUrl } = getBaseUrl())
       entry = blacklistedIPMapping.get(nodeIpPort)
-      if (entry !== undefined) console.log('The ip address blacklisted is', nodeIpPort)
+
+      if (entry !== undefined) {
+        console.log('The IP address blacklisted is', nodeIpPort)
+        // Reassign nodeIpPort and baseUrl to find a new pair
+        ;({ nodeIpPort, baseUrl } = getBaseUrl())
+      }
```

Comment:
Maybe we should exit the loop after a few retries to prevent an infinite loop condition and proceed with the last selected node. Something like:

retries++
      if (retries >= maxRetries) {
        console.error('Failed to find a non-blacklisted IP after max retries.')
        break
      }

Reply:
hmm yeah that makes sense